src/lib/OpenEXRCore/openexr_context.h

```diff
@@ -125,8 +125,10 @@ typedef int64_t (*exr_query_size_func_ptr_t) (
  * truly a stream, it is up to the provider to implement appropriate
  * caching of data to give the appearance of being able to seek/read
  * atomically.
+ *
+ * The return is the number of bytes read.
  */
-typedef int64_t (*exr_read_func_ptr_t) (
```

Comment:
read returns ssize_t, don't think we should change this.

Reply:
also, this is a matter of public interface now, we will need to think about abi compliance / compatibility with other people who have implemented this function. Willing to entertain this, and make pass prior to us saying "this is the stable version" (and I'll need to inject a proper symbol map to handle future symbol versions, now that I know how to do that). Although even with that said, I'm more likely to argue for signed values instead of unsigned..

Reply:
ABI stability aside (which may trump all else if you wish), there are two conflicting methodologies, either of which seems principled:

* Always match what read() returns, even if it differs from platform to platform and may therefore require casting on the user side to be handled uniformly.
* Return a concrete type that is well understood, the same on all platforms and requires no user-side casting, and is big enough to handle the read return on all platforms.

Reply:
I'll have another look at this. The reason I changed it was trying to reconcile the return type of a function being assigned to read_fn. I suppose you were intending to use < 0 as an indication of a problem.

Reply:
Ah right, Larry references posix read, which returns ssize_t, and uses -1 to signal an error.